executor/write.go

```diff
@@ -169,6 +169,18 @@ func updateRecord(ctx context.Context, sctx sessionctx.Context, h kv.Handle, old
 		txnCtx := sctx.GetSessionVars().TxnCtx
 		if txnCtx.IsPessimistic {
 			txnCtx.AddUnchangedRowKey(unchangedRowKey)
+			// Update DeltaForTable
+			colSize := make(map[int64]int64, len(newData))
+			for id, col := range t.Cols() {
+				_, err := codec.EstimateValueSize(sc, newData[id])
+				if err != nil {
+					continue
+				}
+				// Nothing changed so size should be 0
+				colSize[col.ID] = 0
+			}
+			udpp := sctx.GetSessionVars().UseDynamicPartitionPrune()
+			txnCtx.UpdateDeltaForTable(t.Meta().ID, physicalID, 0, 0, colSize, udpp)
```

Comment:
Seems not needed, the keys are already locked in `SelectLockExec`, `PointGetExec`, `BatchPointGetExec` and table ids are recorded there. If there are no keys locked, and no rows changed, which means there will be nothing in memory buffer, the txn will not be marked as `dirty` so the schema check is not done. Some test cases for the `rows changed but not locked` and `no rows changend and locked` could be added for it.

Reply:
Yes, no need for there.